src/kubernetes/state/instrumenting.rs

```diff
@@ -136,19 +136,16 @@ mod tests {
     /// Guarantees only one test will run at a time.
     /// This is required because we assert on a global state, and we don't
     /// want interference.
-    fn tests_lock() -> MutexGuard<'static, ()> {
+    async fn tests_lock() -> MutexGuard<'static, ()> {
         static INSTANCE: OnceCell<Mutex<()>> = OnceCell::new();
-        INSTANCE
-            .get_or_init(|| Mutex::new(()))
-            .lock()
-            .unwrap_or_else(|err| err.into_inner())
+        INSTANCE.get_or_init(|| Mutex::new(())).lock().await
```

Comment:
Looks fine, but why is this required?

Reply:
Clippy rule [await_holding_lock](https://rust-lang.github.io/rust-clippy/master/index.html#await_holding_lock):
```
error: this MutexGuard is held across an 'await' point. Consider using an async-aware Mutex type or ensuring the MutexGuard is dropped before calling await.
   --> src/kubernetes/state/instrumenting.rs:151:13
    |
151 |         let _guard = tests_lock();
    |             ^^^^^^
    |
note: these are all the await points this lock is held through
   --> src/kubernetes/state/instrumenting.rs:151:9
    |
151 | /         let _guard = tests_lock();
152 | |
153 | |         let (mut writer, mut events_rx, mut actions_tx) = prepare_test();
154 | |
...   |
175 | |         join.await.unwrap();
176 | |     }
    | |_____^
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#await_holding_lock
```